provider/gandi/client.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gandi
+
+import (
+	"github.com/go-gandi/go-gandi/domain"
+	"github.com/go-gandi/go-gandi/livedns"
+)
+
+type DomainClientAdapter interface {
+	ListDomains() (domains []domain.ListResponse, err error)
+}
+
+type DomainClient struct {
+	Client *domain.Domain
+}
+
+func (p *DomainClient) ListDomains() (domains []domain.ListResponse, err error) {
+	return p.Client.ListDomains()
+}
+
+func NewDomainClient(client *domain.Domain) DomainClientAdapter {
+	return &DomainClient{client}
+}
+
+// StandardResponse copied from go-gandi/internal/gandi.go
+type StandardResponse struct {
```

Comment:
I would suggest to lower-case all structs, this should only used withing the provider gandi, no need to expose it

Reply:
Ah neat, learning something new every time I use go ;)